subprojects/logging/src/main/java/org/gradle/internal/logging/events/LogEvent.java

```diff
@@ -51,4 +62,34 @@ public void render(StyledTextOutput output) {
     public String toString() {
         return "[" + getLogLevel() + "] [" + getCategory() + "] " + message;
     }
+
+    public static class Builder {
```

Comment:
We need to keep garbage to a minimum in the logging infrastructure. I would not use a builder for these events.